tensornetwork/block_sparse/initialization.py

```diff
@@ -0,0 +1,163 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from tensornetwork.block_sparse.index import Index
+from tensornetwork.block_sparse.blocksparsetensor import _randn, _random
+from tensornetwork.block_sparse.blocksparsetensor import BlockSparseTensor
+from typing import Tuple, Type, Optional, Sequence
+
+
+def ones(indices: Sequence[Index],
+         dtype: Optional[Type[np.number]] = None) -> BlockSparseTensor:
+  """
+  Initialize a symmetric tensor with ones.
+  Args:
+    indices: List of `Index` objecst, one for each leg.
+    dtype: An optional numpy dtype. The dtype of the tensor
+  Returns:
+    BlockSparseTensor
+  """
+
+  return BlockSparseTensor.ones(indices, dtype)
+
+
+def zeros(indices: Sequence[Index],
+          dtype: Optional[Type[np.number]] = None) -> BlockSparseTensor:
+  """
+  Initialize a symmetric tensor with zeros.
+  Args:
+    indices: List of `Index` objecst, one for each leg.
+    dtype: An optional numpy dtype. The dtype of the tensor
+  Returns:
+    BlockSparseTensor
+  """
+
+  return BlockSparseTensor.zeros(indices, dtype)
+
+
+def randn(indices: Sequence[Index],
+          dtype: Optional[Type[np.number]] = None) -> BlockSparseTensor:
+  """
+  Initialize a random symmetric tensor from random normal distribution.
+  Args:
+    indices: List of `Index` objecst, one for each leg.
+    dtype: An optional numpy dtype. The dtype of the tensor
+  Returns:
+    BlockSparseTensor
+  """
+
+  return BlockSparseTensor.randn(indices, dtype)
+
```

Comment:
Is there any reason that all of the initializations should not be moved into the `initialization.py` file? (i.e. instead of having some of them such as `randn` still defined in `BlockSparseTensor`)

Reply:
initialization should now actually all be in `initialization.py` (except classmethod initialization).